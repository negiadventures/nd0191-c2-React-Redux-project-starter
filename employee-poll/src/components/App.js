import React from "react";
import { useEffect, Fragment } from "react";
import { connect } from "react-redux";
import { handleInititaldata } from "../actions/shared";
import Home from "./Home";
import { LoadingBar } from "react-redux-loading-bar";
import NewQuestion from "./NewQuestion";
import UnansweredQuestion from "./UnansweredQuestion";
import { Route, Routes } from "react-router-dom";
import Nav from "./Nav";
import Login from "./Login";
import Leaderboard from "./Leaderboard";

const App = (props) => {
  useEffect(() => {
    props.dispatch(handleInititaldata());
  }, [props]);
  return (
    <Fragment>
      <LoadingBar />

      {props.loading === true ? (
        <Login />
      ) : (
        <div className="container">
          <Nav />
          <Routes>
            <Route path="/" exact element={<Home />}></Route>
            <Route path="/leaderboard" element={<Leaderboard />}></Route>
            <Route
              path="/question/:id"
              element={<UnansweredQuestion />}
            ></Route>
            <Route path="/add" element={<NewQuestion />}></Route>
          </Routes>
        </div>
      )}
    </Fragment>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  loading: authedUser === null,
});

export default connect(mapStateToProps)(App);
