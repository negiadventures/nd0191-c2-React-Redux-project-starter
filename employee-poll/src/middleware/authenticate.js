const authenticate = (store) => (next) => (action) => {
  const returnValue = next(action);
  console.log("auth:", store.getState());
  if(store.getState().authedUser ===''){
    console.log('error')
  }
  return returnValue;
};

export default authenticate;
