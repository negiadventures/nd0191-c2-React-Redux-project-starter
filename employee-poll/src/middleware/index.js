import logger from "./logger";
import thunk from "redux-thunk";
import authenticate from "./authenticate";
import { applyMiddleware } from "redux";

export default applyMiddleware(thunk, authenticate, logger);
