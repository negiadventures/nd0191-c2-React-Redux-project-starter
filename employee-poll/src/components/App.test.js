import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import App from "./App";
import { BrowserRouter as Router } from "react-router-dom";
import reducer from "../reducers";
import middleware from "../middleware";
import { configureStore } from "@reduxjs/toolkit";
const store = configureStore({
  reducer: reducer,
  middleware: middleware,
});
test("renders login page", () => {
  const component = render(
    <Provider store={store}>
      <Router>
        <App />
      </Router>
    </Provider>
  );
  expect(component.getByText(/Employee/i)).toBeInTheDocument();
  expect(component).toMatchSnapshot();
});
