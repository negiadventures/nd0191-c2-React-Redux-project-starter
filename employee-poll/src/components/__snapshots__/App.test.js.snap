// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div
        class="login-container"
      >
        <h1
          class="center"
        >
          Employee Polls
        </h1>
        <div
          class="login-image-container"
        >
          <img
            alt="employee-poll"
            border="0"
            src="https://i.ibb.co/f8vfTmB/employee-poll.jpg"
          />
        </div>
        <h3
          class="center"
        >
          Log In
        </h3>
        <div
          class="login-form"
        >
          <form>
            <label>
              User
            </label>
            <select
              class="input-text"
              data-testid="username"
              id="username"
              name="username"
            >
              <option
                name="username"
              >
                Select a User
              </option>
              <option
                value="sarahedo"
              >
                sarahedo
              </option>
              <option
                value="tylermcginnis"
              >
                tylermcginnis
              </option>
              <option
                value="mtsamis"
              >
                mtsamis
              </option>
              <option
                value="zoshikanlu"
              >
                zoshikanlu
              </option>
            </select>
            <label>
              Password
            </label>
            <input
              class="input-text"
              data-testid="password"
              name="password"
              type="text"
              value=""
            />
            <label />
            <center>
              <button
                class="login-button"
                data-testid="submit-button"
              >
                Submit
              </button>
            </center>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div
      class="login-container"
    >
      <h1
        class="center"
      >
        Employee Polls
      </h1>
      <div
        class="login-image-container"
      >
        <img
          alt="employee-poll"
          border="0"
          src="https://i.ibb.co/f8vfTmB/employee-poll.jpg"
        />
      </div>
      <h3
        class="center"
      >
        Log In
      </h3>
      <div
        class="login-form"
      >
        <form>
          <label>
            User
          </label>
          <select
            class="input-text"
            data-testid="username"
            id="username"
            name="username"
          >
            <option
              name="username"
            >
              Select a User
            </option>
            <option
              value="sarahedo"
            >
              sarahedo
            </option>
            <option
              value="tylermcginnis"
            >
              tylermcginnis
            </option>
            <option
              value="mtsamis"
            >
              mtsamis
            </option>
            <option
              value="zoshikanlu"
            >
              zoshikanlu
            </option>
          </select>
          <label>
            Password
          </label>
          <input
            class="input-text"
            data-testid="password"
            name="password"
            type="text"
            value=""
          />
          <label />
          <center>
            <button
              class="login-button"
              data-testid="submit-button"
            >
              Submit
            </button>
          </center>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
